### PR TITLE
Fixes for nGraph reference implementation of Interpolate-4.

### PR DESCRIPTION
**Pull request by vgavrilo**
Fixing bugs in nGraph reference implementation of Interpolate-4

Issue #39093

### CODE REVIEW

File: ngraph/core/reference/src/runtime/reference/interpolate.cpp [hunk 1]
```diff
@@ -39,18 +39,19 @@ std::array<float, 4> InterpolateEvalHelper::get_cubic_coeff(float s, float a)
...
-        float in_coord = m_get_original_coord(static_cast<float>(output_coord[axis]),
-                                              m_scales[i],
-                                              static_cast<float>(m_out_shape[axis]),
+        float length_original = static_cast<float>(m_input_data_shape[i]);
+        float in_coord = m_get_original_coord(static_cast<float>(output_coord[i]),
+                                              m_all_scales[i],
+                                              static_cast<float>(m_out_shape[i]),
                                               length_original);
-        int64_t nearest_pixel = m_get_nearest_pixel(in_coord, m_scales[i] < 1.0);
-        input_coord[axis] = clip_coord(nearest_pixel, length_original);
```
Comment (ilyachur) [inline ngraph/core/reference/src/runtime/reference/interpolate.cpp:53]: Looks like we can remove `clip_coord` method from the header.

File: ngraph/core/reference/include/ngraph/runtime/reference/interpolate.hpp [hunk 2]
```diff
@@ -226,6 +222,16 @@ namespace ngraph
                     , m_out_shape{out_shape}
                     , m_scales{scales}
                 {
+                    size_t input_rank = input_data_shape.size();
+                    std::vector<float> all_scales(input_rank, 1.0f);
+                    size_t num_of_axes = axes.size();
+
+                    for (size_t i = 0; i < num_of_axes; ++i)
+                    {
+                        all_scales[axes[i]] = scales[i];
```
Comment (ilyachur) [inline ngraph/core/reference/include/ngraph/runtime/reference/interpolate.hpp:231]: Please add the check that `scales` vector has the same size as `axes`.

File: ngraph/core/reference/include/ngraph/runtime/reference/interpolate.hpp [hunk 3]
```diff
@@ -226,6 +222,16 @@ namespace ngraph
                     , m_out_shape{out_shape}
                     , m_scales{scales}
                 {
+                    size_t input_rank = input_data_shape.size();
+                    std::vector<float> all_scales(input_rank, 1.0f);
```
Comment (ilyachur) [inline ngraph/core/reference/include/ngraph/runtime/reference/interpolate.hpp:226]: why we need this local vector? Looks like we can write data to `m_all_scales` without temporary vectors.

**Review by ilyachur — CHANGES_REQUESTED** [review]
I didn't see any tests which check this code. Please add it.

Why do you disable some tests for CPU plugin? Should we merge this functionality if it disables some tests?!

@lazarevevgeny Do we still have a plan to have these changes in release? Will we have changes for CPU? Maybe should we disable Interpolate reference implementation for this release?

**Comment by vgavrilo** [comment]
> I didn't see any tests which check this code. Please add it.
> 
> Why do you disable some tests for CPU plugin? Should we merge this functionality if it disables some tests?!
> 
> @lazarevevgeny Do we still have a plan to have these changes in release? Will we have changes for CPU? Maybe should we disable Interpolate reference implementation for this release?

I disabled CPU tests, because there are differences in results from CPU implementation of Interpolate-4 and results from nGraph reference implementation of Interpolate-4.

File: ngraph/core/reference/include/ngraph/runtime/reference/interpolate.hpp (hunk 3 above)
Comment (vgavrilo) [reply ngraph/core/reference/include/ngraph/runtime/reference/interpolate.hpp]: Done.

File: ngraph/core/reference/include/ngraph/runtime/reference/interpolate.hpp (hunk 2 above)
Comment (vgavrilo) [reply ngraph/core/reference/include/ngraph/runtime/reference/interpolate.hpp]: Done.

**Comment by ilyachur** [comment]
> > I didn't see any tests which check this code. Please add it.
> > Why do you disable some tests for CPU plugin? Should we merge this functionality if it disables some tests?!
> > @lazarevevgeny Do we still have a plan to have these changes in release? Will we have changes for CPU? Maybe should we disable Interpolate reference implementation for this release?
> 
> I disabled CPU tests, because there are differences in results from CPU implementation of Interpolate-4 and results from nGraph reference implementation of Interpolate-4.

You have disabled the tests but you didn't add new to check that reference implementation works correct. 
Also what is the solution for disabled tests for 2021.1 release? Do we want to fix them on CPU side?

Do we really need reference implementation for Interpolate in this release?

File: ngraph/core/reference/src/runtime/reference/interpolate.cpp (hunk 1 above)
Comment (vgavrilo) [reply ngraph/core/reference/src/runtime/reference/interpolate.cpp:53]: Done.

**Comment by vgavrilo** [comment]
> > > I didn't see any tests which check this code. Please add it.
> > > Why do you disable some tests for CPU plugin? Should we merge this functionality if it disables some tests?!
> > > @lazarevevgeny Do we still have a plan to have these changes in release? Will we have changes for CPU? Maybe should we disable Interpolate reference implementation for this release?
> > 
> > 
> > I disabled CPU tests, because there are differences in results from CPU implementation of Interpolate-4 and results from nGraph reference implementation of Interpolate-4.
> 
> You have disabled the tests but you didn't add new to check that reference implementation works correct.
> Also what is the solution for disabled tests for 2021.1 release? Do we want to fix them on CPU side?
> 
> Do we really need reference implementation for Interpolate in this release?

These tests have already been fixed in CPU side (PR https://github.com/openvinotoolkit/openvino/pull/2152).

**Comment by ilyachur** [comment]
> > > > I didn't see any tests which check this code. Please add it.
> > > > Why do you disable some tests for CPU plugin? Should we merge this functionality if it disables some tests?!
> > > > @lazarevevgeny Do we still have a plan to have these changes in release? Will we have changes for CPU? Maybe should we disable Interpolate reference implementation for this release?
> > > 
> > > 
> > > I disabled CPU tests, because there are differences in results from CPU implementation of Interpolate-4 and results from nGraph reference implementation of Interpolate-4.
> > 
> > 
> > You have disabled the tests but you didn't add new to check that reference implementation works correct.
> > Also what is the solution for disabled tests for 2021.1 release? Do we want to fix them on CPU side?
> > Do we really need reference implementation for Interpolate in this release?
> 
> These tests have already been fixed in CPU side (PR #2152).

But as I see the target branch for PR #2152 is master branch. Should we merge this PR to release branch too? What is about the merge process? What is the order of merge these PRs and who should enable these tests?

**Pull request closed without merging**